### DISCUSSION TITLE
how to setup my own rustdesk server?

### DISCUSSION THREAD

**Discussion by ardeal** [rustdesk/rustdesk#4262] [Q&A]
Hi,

I couldn't open the link in https://github.com/rustdesk/rustdesk, [set up your own](https://rustdesk.com/server)

I have already run rustdesk server in my docker, my questions are:
1) should map the port for rustdesk?
2) what configuration should set in rustdesk server?
3) how to connect my rustdesk client to my rustdesk server?

**Comment by Lvdwardt** [comment]
Hey @ardeal!

Here is my configuration for rustdesk and nginx proxy manager:
https://gist.github.com/Lvdwardt/707d9c3fb4581d38102399f91a21c9c6

I'm using a custom domain that gets forwarded to the nginx proxy manager IP; so rustdesk.mydomain.com. In rustdesk client, you have to enter this under ID Server. 
<img width="505" alt="Screenshot 2023-06-05 at 23 32 25" src="https://github.com/rustdesk/rustdesk/assets/87395969/37b41850-d9da-4936-af4a-d7771fd580e7">

That's all I needed to do to start.

Hope this helps!

**Reply by SlalomJohn** [reply]
> Hi,
> 
> I couldn't open the link in https://github.com/rustdesk/rustdesk, [set up your own](https://rustdesk.com/server)
> 
> I have already run rustdesk server in my docker, my questions are:
> 
> 1. should map the port for rustdesk?
> 2. what configuration should set in rustdesk server?
> 3. how to connect my rustdesk client to my rustdesk server?

Client binary .EXE rename as
rustdesk-host=YOUR_SERVER_IP_OR_NAME,key=PUB_KEY_TEXT.exe

sample
rustdesk-host=rd.microsoft.com,key=N445gfAAbBbbds6p0tIs=.exe

Yoy can see this parameters in client About window
